person.rb

```diff
@@ -22,6 +25,11 @@ def can_use_services?
     of_age? || @parent_permission
   end
 
+  def add_rental(date, book)
+    rental = Rental.new(date, book, self)
+    @rentals.push(rental) unless @rentals.include?(rental)
+  end
```

Comment:
- [ ] [OPTIONAL] Kindly remove the line for adding the new Rental to the `@rentals` as you already add it in the Rental class initialize method. This will be marked as optional as you checked if the new rental exists in the `@rentals` list before or not, so it will not affect your code and it will be added once to the `@rentals` list.

Reply:
Ok, thanks for the clarification.